src/main/frontend/src/components/Header.tsx

```diff
@@ -0,0 +1,72 @@
+import {useDispatch, useSelector} from "react-redux";
+import {RootState} from "../store";
+import {Button, PageHeader} from "antd";
+import {useNavigate} from "react-router-dom";
+import {AuthState, logout, updateAccessToken, updateAuth} from "../slices/authSlice";
+import axios from "axios";
+import {useEffect, useState} from "react";
+import UserMenu from "./UserMenu";
+
+export interface UserInfo {
+    id: number | null,
+    name: string | null,
+    imageUrl: string | null
+}
+
+const Header = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+    dispatch(updateAuth());
+
+    const auth = useSelector((state: RootState) => state.auth);
+    const [userInfo, setUserInfo] = useState<UserInfo>({id: null, name: null, imageUrl: null});
+    const [authState, setAuthState] = useState<AuthState>(auth);
+
+    useEffect(() => {
+        if (authState.accessToken !== null) { //login 한 경우
```

Comment:
JS에서는 이렇게도 표현이 가능합니다. null, undefined, empty string 모두 검증할 수 있어요
```suggestion
        if (authState.accessToken) { 
```

Reply:
```suggestion
        if (!authState.accessToken) {
          return;
        }
        
        ...
```

이런식으로 검증후 리턴방식을 쓰면 if블록이 훨씬 작아지게 됩니다.